llvm/docs/TestingGuide.rst

```diff
@@ -568,7 +568,8 @@ list of boolean expressions. The values in each expression may be:
 | ``UNSUPPORTED`` disables the test if any expression is true.
 | ``XFAIL`` expects the test to fail if any expression is true.
 
-As a special case, ``XFAIL: *`` is expected to fail everywhere.
+As a special case, ``XFAIL: *`` is expected to fail everywhere. Similarly,
+``UNSUPPORTED: target={{.*}}`` will disable the test everywhere.
```

Comment:
It might be a good idea to expand on this a bit and explain that `XFAIL` runs the test but expects it to fail, where as `UNSUPPORTED` will skip the test entirely so you would never know if the state changes.

Reply:
Coincidentally, this is documented exactly as you mentioned in https://github.com/llvm/llvm-project/blob/main/llvm/docs/TestingGuide.rst?plain=1#L596

Reply:
I'm not sure `UNSUPPORTED: target={{.*}}` is really a "special case": it's just a natural consequence of the use of a regex with the `target=` method. `XFAIL: *` on the other hand IS a special case, because it isn't a regex.

Reply:
yeah, it is however useful to disable the test in all cases. lmk if you think this is still an interesting thing to document.

Reply:
@jh7370 I agree with you, but I guess I didn't read it as "this is also a special case", but rather as "this is another way to do this". I think that having the pointer to the fact that this is not great is important since we don't want to encourage people to disable tests I think.

Reply:
How about we rephrase this as: "To mark a test expected failing or unsupported for all cases, use `XFAIL: *` or `UNSUPPORTED: target={{.}}` as appropriate" or something to that effect (most importantly, simply drop the "as a special case" bit)?

I've definitely seen a use-case downstream for marking something as `UNSUPPORTED` for all cases: if a test from upstream has come down and is flaky (so doesn't pass/fail consistently and thus `XFAIL` won't work), it's important to mark it as such whilst it gets reported upstream, so that our continuous merge process doesn't get blocked.

Reply:
I like that suggestion @jh7370

Reply:
> I've definitely seen a use-case downstream for marking something as UNSUPPORTED for all cases

Same. it took me a bit to find this trick with UNSUPPORTED.